### PR TITLE
525 re design thyra home page

### PR DESCRIPTION
**Pull request by 0xMazout**

### CODE REVIEW

**Comment by 0xMazout** [comment]
I'm spending time debugging the CI

**Comment by 0xMazout** [comment]
image

The ci is blocking on this and in my local repo i have no problem.

File: web/home/src/components/managePluginCard.tsx
Comment (Thykof) [inline web/home/src/components/managePluginCard.tsx:7]: did you forgot to run prettier?

File: web/home/src/pages/Home/home.tsx
Comment (Thykof) [inline web/home/src/pages/Home/home.tsx:33]: what is the type of variable `wallet` here? how the plugin is supposed to pass the logo?

File: web/home/src/pages/Home/home.tsx
Comment (Thykof) [inline web/home/src/pages/Home/home.tsx:106]: why don't you add home attribute to PluginHomePage, so that you don't need useState<PluginHome?

File: web/plugin-manager/src/components/toggleTheme.tsx
Comment (Thykof) [inline web/plugin-manager/src/components/toggleTheme.tsx:20]: duplicated code here, with another react project, is there a way to reuse component from one project to another?

**Review by Thykof — CHANGES_REQUESTED** [review]

File: web/home/src/pages/Home/home.tsx
Comment (0xMazout) [reply web/home/src/pages/Home/home.tsx:106]: Yes it was a pure mistake, i realized that on the Plugin manager pr  and i changed it in the PR of plugin manager page.

File: web/plugin-manager/src/components/toggleTheme.tsx
Comment (0xMazout) [reply web/plugin-manager/src/components/toggleTheme.tsx:20]: I can't send it to the shared folder because it's JSX. so for now it's okay to duplicate because it's generic. But yes we will merge to one folder when we rebrand (Soon)

**Review by Thykof — DISMISSED** [review]

File: web/home/src/pages/Home/home.tsx
Comment (mazmassa) [inline web/home/src/pages/Home/home.tsx:107]: Non blocking: I would have these both status on a constants or in a config JSON file. If the BE changes we can update straight forward.

```js 
export const PLUGIN = {
    UP: "up",
    STARTING: "starting"
}
```
then you can use:

```jsx
if (plugin.status == PLUGIN.STARTING || plugin.status == PLUGIN.UP){...}
```

File: web/home/src/pages/Home/home.tsx
Comment (mazmassa) [inline web/home/src/pages/Home/home.tsx:79]: I think our headers would be always with **json** content type, isn't it!? 🤔 I am not sure using axios but I think it will work by default. It  wouldn't work if you remove the header conf?

In any case, I would place this request in a specific file for REST actions. Specially if we need reuse that in future. Specially if we need to check for errors, wrapping that in a try/catch, etc... it can be hard to scale if we keep it here in the same place.

File: web/home/src/components/managePluginCard.tsx
Comment (Thykof) [reply web/home/src/components/managePluginCard.tsx:21]: @SlnPons cc

Comment (SlnPons) [reply web/home/src/components/managePluginCard.tsx:21]: It would make sense to do what you suggest Mario in the next version of the home page when implementing the rebranding (depending on the work evaluation).

File: web/home/src/pages/Home/home.tsx
Comment (0xMazout) [reply web/home/src/pages/Home/home.tsx:103]: Yes i agree a lot will upgrade my knowledge on React Query to implement it in our . At the beginning i was using React Query but i was too noob on it 🙃 . But will spend some time to reintroduce it on the rebranding Thanks 😄

File: web/plugin-manager/src/components/Header.tsx
Comment (0xMazout) [reply web/plugin-manager/src/components/Header.tsx:10]: Noted and will be introduced in the rebranding 👍

File: web/home/src/components/managePluginCard.tsx
Comment (0xMazout) [reply web/home/src/components/managePluginCard.tsx:21]: Yes we will try to implement the internationalisation in the rebranding phase and add more robustness

File: web/home/src/components/pluginCard.tsx
Comment (0xMazout) [reply web/home/src/components/pluginCard.tsx:21]: Noted and will be implemented on the rebranding phase with Twin Macro . I think we can create Styled components and increase the reusability with clean components 👍

File: web/home/src/pages/Home/home.tsx
Comment (0xMazout) [reply web/home/src/pages/Home/home.tsx:79]: Yes clearly , i think we can share the header to all request and regroup this into axios.ts as it's done in Plugin Manager page. 

At the rebranding we plan to merge Homepage and PluginManagerPage in one project and we will use the axios.ts from plugin Manager Page. So we can accept it will be solve at the merge of the two projects 😃

File: web/home/src/index.css
Comment (0xMazout) [reply web/home/src/index.css]: For the knowledge, We use https://socket.dev/npm/package/tw-colors 
for theming 
And the way to use the colors defined in differents themes is   
our-class{
 background-color: hsl(var(--twc-bgCard));
}

File: web/home/src/pages/Home/home.tsx
Comment (SlnPons) [reply web/home/src/pages/Home/home.tsx:104]: How do we make sure not to forget things we should while rebranding? Can you create a doc or to do list somewhere please?

**Comment by 0xMazout** [comment]
Lots of coments are adressed in this issue : https://github.com/massalabs/thyra/issues/556

File: web/home/src/components/managePluginCard.tsx
Comment (mazmassa) [reply web/home/src/components/managePluginCard.tsx:21]: Let's do it together! I'll let you ping me when you want. We start our base for that and when necessary we upgrade for something bigger. We can do better for now with few lines of code. Please, ping me.

Closing here meanwhile! :)

File: web/plugin-manager/src/components/Header.tsx
Comment (mazmassa) [reply web/plugin-manager/src/components/Header.tsx:10]: Sure! :)

**Review by mazmassa — APPROVED** [review]
Approving here since we agreed in make the changes (some together) ASAP.

**Pull request merged**
